### ISSUE TITLE
Testing before deploying

### ISSUE THREAD

**Issue by es20490446e** [linuxdeepin/developer-center#745]
In the latest months many various bugs on Deepin have appeared, which I chose to report.

But thinking deeper about it, it simply isn't worth the effort. Those bugs are simply obvious by manually testing the software before pushing the changes to GitHub. So I will no longer report those, and just stick to the non so obvious ones.

Also realize that automate testing is quite useful for making sure that a software still works, and that it still plays nice with the external software that it depends upon.

But outside of that it does very little, test-cases won't cover many non fatal bugs and trying to do so is quite time expensive. It's just simpler to immediately manually test changes than writing a test case for every commit, or expect that bugs will be fixed some time in the future.

More broadly the idea I'm suggesting is to fully complete a feature, warranting its quality, before continuing with new ones. So to reduce the amount of half baked components. Aka [One Piece Flow](https://www.kaizenworld.com/_Media/slide1-3-2.jpeg).

**Comment by hualet** [comment]
Hi @es20490446e , sorry for the delay of this reply.

We've been trying rolling releases in the last 1~2 months, and we worked very hard on a lot of new features, trying to make the community know that we're passionate on this project and encourage others to do so, but also I have to admit it that we pushed it too hard. so I'm considering slowing down a little bit , ensuring the quality, also put some efforts of the devs in maintaining the communication between the community and us.

Hope that will make a difference, cheers.
